uaa_bot/client.py

```diff
@@ -0,0 +1,196 @@
+from posixpath import join as urljoin
+import json
+import time
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+from uaa_bot import config
+
+
+class UAAError(RuntimeError):
+    """
+    This exception is raised when the UAA API returns a status code >= 400
+    Attributes:
+        response:   The full response object from requests that was returned
+        error:  The body of the response json decoded
+    Args:
+        response: The full response object that is causing this exception to be raised
+    """
+
+    def __init__(self, response):
+        self.response = response
+        self.error = json.loads(response.text)
+
+        message = self.error["error_description"]
+
+        super(UAAError, self).__init__(message)
+
+
+class UAAClient(object):
```

Comment:
No need for `(object)` here. :-)

```suggestion
class UAAClient:
```